writer.go

```diff
@@ -0,0 +1,246 @@
+// Copyright (C) 2022 by Posit Software, PBC
+package rsf
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+type writer struct {
```

Comment:
Can these be named something like `rsfWriter`/`rsfReader` since they are specific implementations of the `Writer`/`Reader` interfaces?

Reply:
Sure.